### PR TITLE
[CORDA-1638]: Audit log of failed transactions MVP.

### PR DESCRIPTION
**Pull request by sollecitom**
https://r3-cev.atlassian.net/browse/ENT-1638

### CODE REVIEW

File: core/src/main/kotlin/net/corda/core/flows/FinalityFlow.kt [hunk 1]
```diff
@@ -51,17 +52,24 @@ class FinalityFlow(val transaction: SignedTransaction,
...
         progressTracker.currentStep = BROADCASTING
-        for (party in parties) {
-            if (!serviceHub.myInfo.isLegalIdentity(party)) {
-                val session = initiateFlow(party)
-                subFlow(SendTransactionFlow(session, notarised))
-            }
+        val recipients = parties.filterNot(serviceHub.myInfo::isLegalIdentity)
+        logger.info("Broadcasting transaction to parties ${recipients.map { it.name }.joinToString(", ", "[", "]")}.")
+        for (party in recipients) {
+            logger.info("Sending transaction to party ${party.name}.")
```
Comment (anvabr) [inline core/src/main/kotlin/net/corda/core/flows/FinalityFlow.kt]: Probably unnecessary, we know who we are sending it to so if there is no 'received' entry it means that one has failed

File: core/src/main/kotlin/net/corda/core/flows/FinalityFlow.kt [hunk 2]
```diff
@@ -73,9 +81,12 @@ class FinalityFlow(val transaction: SignedTransaction,
             val notarySignatures = subFlow(NotaryFlow.Client(transaction))
             transaction + notarySignatures
         } else {
+            logger.info("No need to notarise this transaction.")
             transaction
         }
+        logger.info("Recording transaction locally.")
```
Comment (anvabr) [inline core/src/main/kotlin/net/corda/core/flows/FinalityFlow.kt]: Also probably not needed as we know what needs to happen and if it does not happen we know it did not.

File: core/src/main/kotlin/net/corda/core/flows/FinalityFlow.kt (hunk 1 above)
Comment (sollecitom) [reply core/src/main/kotlin/net/corda/core/flows/FinalityFlow.kt]: As discussed, good point, but this lowers the chance a flow might get killed in between, in which case this information would help.

File: core/src/main/kotlin/net/corda/core/flows/FinalityFlow.kt (hunk 2 above)
Comment (sollecitom) [reply core/src/main/kotlin/net/corda/core/flows/FinalityFlow.kt]: As above.

**Pull request merged**